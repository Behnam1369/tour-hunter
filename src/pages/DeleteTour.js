import React from 'react';
import Button from '../components/Button';

const DeleteTour = () => {
  const tours = [
    {
      id: 1,
      title: 'Tour 1',
    },
    {
      id: 2,
      title: 'Tour 1',
    },
    {
      id: 3,
      title: 'Tour 1',
    },
    {
      id: 4,
      title: 'Tour 1',
    },
    {
      id: 5,
      title: 'Tour 1',
    },
  ];

  const deleteTour = (e) => {
    e.preventDefault();
  };

  return (
    <section>
      <h1>Delete a Tour</h1>
      <div>
        {tours.map((tour) => (
          <form key={tour.id} onSubmit={deleteTour}>
            <span>{tour.id}</span>
            <p>{tour.title}</p>
            <Button btnName="Delete" btnType="submit" bgColor="bg-red text-white" />
          </form>
        ))}
      </div>
    </section>
  );
};

export default DeleteTour;
