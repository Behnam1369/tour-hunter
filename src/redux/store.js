import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import usersReducer from './reducers/users';
import reservationReducer from './reducers/reservation';
import tokenReducer from './reducers/token';

const store = configureStore({
  reducer: {
    signedIn: usersReducer,
    reservation: reservationReducer,
    token: tokenReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
