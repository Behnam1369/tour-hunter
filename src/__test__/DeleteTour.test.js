import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import DeleteTour from '../pages/DeleteTour';

describe('Test Admin Delete Tour', () => {
  it('should render the page', () => {
    const element = (
      <Provider store={store}>
        <DeleteTour />
      </Provider>
    );
    const deleteTour = renderer.create(element);
    expect(deleteTour).toMatchSnapshot();
  });
});
