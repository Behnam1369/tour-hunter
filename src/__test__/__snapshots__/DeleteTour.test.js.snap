// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Admin Delete Tour should render the page 1`] = `
<div
  className="container relative mx-auto p-3 w-screen"
>
  <h1
    className="text-center pt-12 text-2xl font-bold"
  >
    Delete a Tour
  </h1>
</div>
`;
